openquake/sep/landslide/nowicki_jessee.py

```diff
@@ -0,0 +1,143 @@
+    "230": -1.08
+}
+
+
+lithology_values={
+    "mt": -1.87,
+    "nd": -0.66,
+    "pa": -0.78,
+    "pb": -1.88,
+    "vi": -1.61,
+    "py": -1.05,
+    "sc": -0.95,
+    "sm": -1.36,
+    "ss": -1.92,
+    "su": -1.36,
+    "va": -1.54,
+    "vb": -1.50,
+    "pi": -0.81
+}
+
+
+LANDCOVER_TABLE = {**landcover_values, **{bytes(k, 'utf-8'): v for k, v in landcover_values.items()}}
+LITHOLOGY_TABLE = {**lithology_values, **{bytes(k, 'utf-8'): v for k, v in lithology_values.items()}}
+
+
+def sigmoid(x):
+    return 1.0 / (1.0 + np.exp(-x))
+
+
+def _landslide_spatial_extent(a: float, b: float, c: float, d: float, p: float):
+    """Calculates the landslide spatial extent (LSE) as per formulae 9
+    from the Reference. LSE after an earthquake can be interpreted as the
+    portion of each cell that is expected to have landslide occurrence.
+
+    Reference: Nowicki Jessee, M. A., Hamburger, M. W., Allstadt, K., 
+    Wald, D. J., Robeson, S. M., Tanyas, H., et al. (2018). 
+    A global empirical model for near-real-time assessment of seismically 
+    induced landslides. 
+    Journal of Geophysical Research: Earth Surface, 123, 1835–1859. 
+    https://doi.org/10.1029/2017JF004494
+    """
+    LSE = np.exp(a + b * p + c * p**2 + d * p**3)
+    return LSE
+
+    
+def nowicki_jessee_2018(
+    pga: Union[float, np.ndarray],
+    pgv: Union[float, np.ndarray],
+    slope: Union[float, np.ndarray],
+    lithology: str,
+    landcover: Union[int, np.ndarray],
+    cti: Union[float, np.ndarray],
+    intercept: float = -6.30,
+    pgv_coeff: float = 1.65,
+    slope_coeff: float = 0.06,
+    coeff_table_lith=LITHOLOGY_TABLE,
+    coeff_table_cov=LANDCOVER_TABLE,
+    cti_coeff: float = 0.03,
+    interaction_term: float = 0.01
+) -> Union[float, np.ndarray]:
+    """
+    Calculates the probability of landsliding using the logistic
+    regression of Nowicki Jessee et al. (2018). As per USGS recommendation,
+    the values of pgv and cti are capped to 211 cm/s and 19, respectively.
+    Furthermore, the model computes the areal coverage, which unbiases the
+    calculated probabilities.
+
+    Reference: Nowicki Jessee, M. A., Hamburger, M. W., Allstadt, K., 
+    Wald, D. J., Robeson, S. M., Tanyas, H., et al. (2018). 
+    A global empirical model for near-real-time assessment of seismically 
+    induced landslides. 
+    Journal of Geophysical Research: Earth Surface, 123, 1835–1859. 
+    https://doi.org/10.1029/2017JF004494
+
+    :param pgv:
+        Peak Ground Velocity, measured in cm/s
+    :param slope:
+        Topographic slope expressed in degrees
+    :param lithology:
+        Rock lithology, a "measure" of rock strength
+    :param landcover:
+        Land cover, procxy for vegetation cover
+    :param cti:
+        Compound Topographic Index, a proxy for soil wetness.
+
+    :returns:
+        prob_ls: Probability of landslide.
+        coverage: Landslide areal coverage.
+    """
+
+    if isinstance(lithology, (str)):
```

Comment:
Do we need the tuple `(str)` or can it be simply `str`? Same for `(int)` on line 120 below.

Reply:
we don't. fixed